com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/ReferencePattern.java

```diff
@@ -20,6 +20,10 @@ private ReferencePattern(String pattern, String patternFormat) {
     protected boolean hasPropertySetSegment() {
         return FULLY_QUALIFIED.equals(this);
     }
+    
+    protected boolean hasDepthOfReference() {

```

Comment:
I have to see how this method is used in PropertiesResolver to understand its name. Wouldn't it be better if it were called `isPlaceholder`? That way the meaning here would be clear and its usage in PropertiesResolver would be able to express more clearly where we could have "deep references".

Furthermore, "normal" `~{references}` can also have depth. See the `FULLY_QUALIFIED` enum value above.